src/link.c

```diff
@@ -724,16 +739,28 @@ LinkTable *LinkTable_disk_open(const char *dirn)
 
     LinkTable *linktbl = CALLOC(1, sizeof(LinkTable));
 
-    fread(&linktbl->num, sizeof(int), 1, fp);
+    if (sizeof(int) != fread(&linktbl->num, sizeof(int), 1, fp)) {
```

Comment:
Yikes, 
> On  success, fread() and fwrite() return the number of items read or written.  This number equals the number of bytes transferred only when size is 1.  If an error occurs, or the end of the file is reached, the return value is a short item count (or zero).

Reply:
This makes me wonder if the cache system has been broken on the master branch since last September...

Reply:
The point is that ``sizeof(int)`` is 4, but we are only expecting ``1`` to be returned here, as we are reading 1 int.